### PR TITLE
Make batch sizes configurable

### LINKED ISSUE #3708: Make batch sizes configurable
### What should be done?

https://github.com/FerretDB/FerretDB/blob/74009c036163fc63840dd6faaae27a084816ea35/internal/handlers/sqlite/msg_insert.go#L88
https://github.com/FerretDB/FerretDB/blob/74009c036163fc63840dd6faaae27a084816ea35/internal/backends/postgresql/collection.go#L133
https://github.com/FerretDB/FerretDB/blob/74009c036163fc63840dd6faaae27a084816ea35/internal/backends/sqlite/collection.go#L117

Currently, both handler-level and backends-level batch sizes are constants. That's because we don't want to consume too much memory on the FerretDB side and go over backend limits such as a maximum number or size of query parameters. We also don't know how to detect and/or auto-tune those parameters yet.

But we should allow changing those values via testing flags.

### Where?

See reference above.

### Definition of Done

- handler updated;
- all backends updated;
- benchmarks updated and run;
- spot refactorings done.

### PR DESCRIPTION
**Pull request by kropidlowsky**
# Description

Replaced batch sizes  hard-cored values with configuratable for: sqlite, postgrsql, hana.

Closes #3708.

## Readiness checklist

- [x] I added/updated unit tests (and they pass).
- [ ] I added/updated integration/compatibility tests (and they pass).
- [x] I added/updated comments and checked rendering.
- [x] I made spot refactorings.
- [ ] I updated user documentation.
- [x] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Milestone (`Next`), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

File: internal/backends/postgresql/collection.go [hunk 1]
```diff
@@ -134,8 +134,7 @@ func (c *collection) InsertAll(ctx context.Context, params *backends.InsertAllPa
 	}
 
 	err = pool.InTransaction(ctx, p, func(tx pgx.Tx) error {
-		// TODO https://github.com/FerretDB/FerretDB/issues/3708
-		const batchSize = 100
+		batchSize := c.r.BatchSize
```
Comment (AlekSi) [inline internal/backends/postgresql/collection.go:137]: Let's panic if batchSize < 1

File: internal/backends/sqlite/collection.go [hunk 2]
```diff
@@ -109,8 +109,7 @@ func (c *collection) InsertAll(ctx context.Context, params *backends.InsertAllPa
 	meta := c.r.CollectionGet(ctx, c.dbName, c.name)
 
 	err := db.InTransaction(ctx, func(tx *fsql.Tx) error {
-		// TODO https://github.com/FerretDB/FerretDB/issues/3708
-		const batchSize = 100
+		batchSize := c.r.BatchSize
```
Comment (AlekSi) [inline internal/backends/sqlite/collection.go:112]: Let's panic if batchSize < 1

**Review by AlekSi — COMMENTED** [review]
Tests fail because we don't pass BatchSize somewhere, and invalid SQL is generated for INSERT

File: cmd/ferretdb/main.go [hunk 3]
```diff
@@ -101,6 +101,7 @@ var cli struct {
 		} `embed:"" prefix:"capped-cleanup-"`
 
 		EnableNewAuth bool `default:"false" help:"Experimental: enable new authentication."`
+		BatchSize     int  `default:"100"   help:"number of maximum size of query parameters"`
```
Comment (chilagrow) [inline cmd/ferretdb/main.go:104]: It's actually for the insertion batch size.

**Review by chilagrow — DISMISSED** [review]
Thanks for your contribution!

**Review by rumyantseva — DISMISSED** [review]
Thanks for contributing!

**Review by AlekSi — APPROVED** [review]

**Comment by AlekSi** [comment]
Thank you!

**Pull request merged**